File: Python3/zigzag-conversion.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if numRows == 1:
            return s

        strings = [""] * numRows
        i = 0
        incremental = 1
        for character in s:
            if i == 0:
                incremental = 1
            elif i == numRows - 1:
                incremental = -1
            strings[i] += character
            i += incremental

        return "".join(strings)

File: Golang/zigzag-conversion.go
import (
    "strings"
)

func convert(s string, numRows int) string {
    if numRows == 1 {
        return s
    }

    stringsArray := make([]string, numRows)
    i := 0
    incremental := 1
    for _, char := range s {
        stringsArray[i] += string(char)
        if i == 0 {
            incremental = 1
        } else if i == numRows - 1 {
            incremental = -1
        }
        i += incremental
    }
    return strings.Join(stringsArray, "")
}
